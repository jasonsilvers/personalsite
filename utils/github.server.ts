import { Octokit as createOctokit } from '@octokit/rest'
import { throttling } from '@octokit/plugin-throttling'

const Octokit = createOctokit.plugin(throttling)

type ThrottleOptions = {
  method: string
  url: string
  request: { retryCount: number }
}
const octokit = new Octokit({
  auth: process.env.BOT_GITHUB_TOKEN,
  throttle: {
    onRateLimit: (retryAfter: number, options: ThrottleOptions) => {
      console.warn(
        `Request quota exhausted for request ${options.method} ${options.url}. Retrying after ${retryAfter} seconds.`
      )

      return true
    },
    onAbuseLimit: (options: ThrottleOptions) => {
      octokit.log.warn(`Abuse detected for request ${options.method} ${options.url}`)
    }
  }
})

async function downloadDirList(path: string) {
  const resp = await octokit.repos.getContent({
    owner: 'jasonsilvers',
    repo: 'personalsite',
    path
  })

  const data = resp.data

  if (!Array.isArray(data)) {
    throw new Error(
      `Tried to download content from ${path}. GitHub did not return an array of files. This should never happen...`
    )
  }

  return data
}

async function downloadDirectory(dir: string): Promise<Array<GitHubFile>> {
  const dirList = await downloadDirList(dir)

  const result = await Promise.all(
    dirList.map(async ({ path: fileDir, type, sha }) => {
      switch (type) {
        case 'file': {
          const content = await downloadFileBySha(sha)
          return { path: fileDir, content }
        }
        case 'dir': {
          return downloadDirectory(fileDir)
        }
        default: {
          throw new Error(`Unexpected repo file type: ${type}`)
        }
      }
    })
  )

  return result.flat()
}

async function downloadFileBySha(sha: string) {
  const { data } = await octokit.request('GET /repos/{owner}/{repo}/git/blobs/{file_sha}', {
    owner: 'jasonsilvers',
    repo: 'personalsite',
    file_sha: sha
  })
  //                                lol
  const encoding = data.encoding as Parameters<typeof Buffer.from>['1']
  return Buffer.from(data.content, encoding).toString()
}

export { downloadDirList }
