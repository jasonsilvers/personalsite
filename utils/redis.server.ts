import Redis from 'ioredis'
import { config } from './misc'

const REDIS_URL = config.redisUrl

export function createClient(): Redis.Redis {
  return new Redis(REDIS_URL)
}


