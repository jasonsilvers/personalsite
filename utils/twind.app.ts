import App from 'next/app'

import { setup } from 'twind'
import { twindConfig } from './twind.config'

export default function withTwindApp<Base = typeof App>(BaseApp?: Base): Base

export default function withTwindApp<Base = typeof App>(BaseApp?: Base): Base

export default function withTwindApp<Base = typeof App>(BaseApp?: Base): Base {
  // If this run on the server _document.js has already done the setup
  if (typeof window !== 'undefined') {
    setup(twindConfig)
  }

  // @ts-ignore
  return BaseApp || App
}
