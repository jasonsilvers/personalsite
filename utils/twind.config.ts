export const twindConfig = {
  theme: {
    fontFamily: {
      sans: 'Raleway, sans-serif'
    }
  },
  preflight: {
    '@import': `url('https://fonts.googleapis.com/css2?family=Raleway:wght@100;400&display=swap')`,
    '@font-face': [
      {
        fontFamily: 'Raleway',
        fontWeight: '100',
        src: 'url(/fonts/Raleway-Thin.tff) format("tff")'
      },
      {
        fontFamily: 'Raleway',
        fontWeight: '400',
        src: 'url(/fonts/Raleway-Regular.tff) format("tff")'
      }
    ]
  }
}
