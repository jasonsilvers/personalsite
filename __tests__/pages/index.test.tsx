import { rtlRender } from '../utils/'
import Home from '../../pages/index'

it('routes the user to the Unauthenticated app', async () => {
  const { getByText, container } = rtlRender(<Home />)

  expect(getByText(/jason/i)).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
