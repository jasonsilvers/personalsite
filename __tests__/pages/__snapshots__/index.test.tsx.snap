// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes the user to the Unauthenticated app 1`] = `
<div>
  <div>
    <header
      class="border-1 py-6"
    >
      <div
        class="tw-1tz6xsh"
      >
        <nav
          class="flex flex-row gap-x-3"
        >
          <h1
            class="text-2xl pr-10 font-thin"
          >
            Jason C. Silvers
          </h1>
          <div
            class="ml-auto flex flex-row gap-x-2"
          >
            <a
              class="h-[32px]"
              href="https://www.linkedin.com/in/jasonsilvers"
              title="Link to linkedin profile"
            >
              Next image stub
            </a>
            <a
              class="h-[32px]"
              href="https://www.linkedin.com/in/jasonsilvers"
              title="Link to linkedin profile"
            >
              Next image stub
            </a>
            <a
              class="h-[32px]"
              href="https://www.strava.com/athletes/297175/training/log"
              title="Link to Strava profile"
            >
              Next image stub
            </a>
            <a
              class="h-[32px]"
              href="https://github.com/jasonsilvers"
              title="Link to github profile"
            >
              Next image stub
            </a>
          </div>
        </nav>
      </div>
    </header>
    <main>
      <div
        class="tw-1tz6xsh pt-6"
      >
        <section
          class="h-[300px] bg-[#F8FAFB] p-8 rounded-2xl flex flex-row"
        >
          <div
            class="flex flex-col gap-y-6"
          >
            <h1
              class="text-2xl"
            >
              Hello
            </h1>
            <p>
              Iâ€™m a web developer.
            </p>
            <p>
              Here, I share my experience as a web developer and things I am learning about on React, Typescript, Next.js, Security and Testing
            </p>
          </div>
          <div
            class="ml-auto"
          >
            Next image stub
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;
